htdocs/product/class/product.class.php

```diff
@@ -1592,6 +1591,9 @@ public function update($id, $user, $notrigger = 0, $action = 'update', $updatety
 			$sql .= ", fk_price_expression = ".($this->fk_price_expression != 0 ? (int) $this->fk_price_expression : 'NULL');
 			$sql .= ", fk_user_modif = ".($user->id > 0 ? $user->id : 'NULL');
```

Comment:
@eldy Old code but - in line with security rules that I see applied for Dolibarr a cast would be needed on '$user->id' but this was not detected with the phpunit check that scans the code.